tests/lang/s02_core/s01_procedures/t02_argument_passing.nim

```diff
@@ -0,0 +1,318 @@
+  ## using named argument passing even regular ordering would've
+  ## prevented this
+
+
+block default_and_regular_argument:
+  proc withDefault(arg1: int, arg2: string = "123"): (int, string) =
+    return (arg1, arg2)
+
+  doAssert withDefault(1) == (1, "123")
+  doAssert withDefault(2, "1") == (2, "1")
+
+block multiple_default_arguments:
+  proc withDefaultSplit(arg1: int = 1, arg2: int = 2): (int, int) =
+    return (arg1, arg2)
+
+  doAssert withDefaultSplit() == (1, 2)
+  doAssert withDefaultSplit(9) == (9, 2)
+  doAssert withDefaultSplit(3, 4) == (3, 4)
+
+
+  proc withDefaultMerged(arg1, arg2: int = 12): (int, int) = (arg1, arg2)
+
+  doAssert withDefaultMerged() == (12, 12)
+  doAssert withDefaultMerged(9) == (9, 12)
+  doAssert withDefaultMerged(3, 4) == (3, 4)
+
+block complex_expression_as_default:
+  block use_other_argument:
+    proc useOther(arg1: int, arg2: int = arg1): (int, int) = (arg1, arg2)
+
+    doAssert useOther(1) == (1, 1)
+    doAssert useOther(1, 2) == (1, 2)
+
+
+  block use_field_of_other_argument:
+    type
+      Object = object
+        field: int
+
+    proc useOther(obj: Object, arg: int = obj.field): (int, int) = (obj.field, arg)
+
+    doAssert useOther(Object(field: 1)) == (1, 1)
+    doAssert useOther(Object(field: 1), 2) == (1, 2)
+
+
+  block call_procedure:
+    proc getValue(): int = 12
+
+    proc useOther(arg: int = getValue()): int = arg
+
+    doAssert useOther(getValue()) == useOther()
+
+    doAssert useOther() == 12
+    doAssert useOther(1) == 1
+
+  block evaluate_block_expression:
+    proc useOther(
+      arg: int = 12,
+      other: int =
+        block:
+          arg
+    ): (int, int) = (arg, other)
+
+    doAssert useOther() == (12, 12)
+    doAssert useOther(2) == (2, 2)
+    doAssert useOther(1, 2) == (1, 2)
+
+
+
+
+block varargs_arguments_no_converter:
+  block single_varargs:
+    proc impl(vas: varargs[int]): seq[int] = @vas
+
+
+    doAssert impl() == newSeq[int]()
+    doAssert impl(1) == @[1]
+    doAssert impl(1, 2, 4, 5, 6, 7) == @[1, 2, 4, 5, 6, 7]
+
+  # Multiple varargs is most likely a feature that is not used widely
+  # (if at all), but this is stil supported, so should probably be tested.
+  block multiple_varargs:
+    proc impl(va1: varargs[int], va2: varargs[bool]): (seq[int], seq[bool]) =
+      return (@va1,  @va2)
+
+    doAssert impl(1) == (@[1], newSeq[bool]())
+
+    if false:
+      # Right nor results in `(@[1, 2, 0], @[true, false, false])`
+      doAssert impl(1, 2, false) == (@[1, 2], @[false])
+
+    ## It is not possible to entirely omit leading varargs arguments, but it
+    ## is possible to explicitly pas values to the other varargs if present
+    doAssert impl(va2 = true) == (newSeq[int](), @[true])
+
+
+  block command_syntax:
+    proc vaProc(args: varargs[int]): seq[int] = @args
+
+    block:
+      ## Command syntax cannot be used in the `let val = cmd 1,2,3,4,4`
+      ##
+      ## REVIEW is it a bug?
```

Comment:
Not sure, could well be

Reply:
Now we need to figure out how to record these so we can easily revisit the inventory of wtfs